ocis-pkg/store/store.go

```diff
@@ -51,32 +32,59 @@ type Options struct {
 // and table to prevent collisions with other microservices.
 // Recommended approach is to use "services" or "ocis-pkg" for the database,
 // and "services/<service-name>/" or "ocis-pkg/<pkg>/" for the package name.
-func Create(opts ...Option) store.Store {
-	options := &Options{}
+func Create(opts ...store.Option) store.Store {
+	options := &store.Options{
+		Context: context.Background(),
+	}
 	for _, o := range opts {
 		o(options)
 	}
 
-	storeopts := storeOptions(options)
+	cacheOpts, ok := options.Context.Value(cacheOptionsContextKey{}).(CacheOptions)
+	if !ok {
+		cacheOpts = CacheOptions{}
+	}
 
-	switch options.Type {
+	switch cacheOpts.Type {
```

Comment:
Why don't we just use `options.Type`? The store should not care if he is used for a cache or something else.

Reply:
because the micros store interface has no Type option. I moved our options to a dedicated StoreOptions option ...